bin/target.ml

```diff
@@ -165,7 +165,24 @@ let resolve_path path ~(setup : Dune_rules.Main.build_system)
     | None -> can't_build path)
 ;;
 
-let expand_path (root : Workspace_root.t) ~(setup : Dune_rules.Main.build_system) ctx sv =
+let expand_path_from_root' (root : Workspace_root.t) sctx ctx sv =
+  let dir =
+    Path.Build.relative
+      ctx.Context.build_dir
+      (String.concat ~sep:Filename.dir_sep root.to_cwd)
+  in
+  let* expander = Action_builder.of_memo (Dune_rules.Super_context.expander sctx ~dir) in
+  let expander = Dune_rules.Dir_contents.add_sources_to_expander sctx expander in
+  let+ s = Dune_rules.Expander.expand_str expander sv in
+  root.reach_from_root_prefix ^ s
+;;
+
+let expand_path_from_root
```

Comment:
Can we somehow unify these functions?

Reply:
Yup, updating the arguments make the `expand_path_from_root'` function obsolete.